pandas/core/arrays/base.py

```diff
@@ -867,6 +894,24 @@ def copy(self) -> ABCExtensionArray:
         """
         raise AbstractMethodError(self)
 
+    def view(self, dtype=None) -> ABCExtensionArray:
+        """
+        Return a view on the array.
+
+        Returns
+        -------
+        ExtensionArray
+
+        Notes
+        -----
+        - This must return a *new* object, not self.
+        - The only case that *must* be implemented is with dtype=None,
+          giving a view with the same dtype as self.
```

Comment:
Can / should we make any requirements on this being no-copy? `i.e.`


```python
a = my_array(...)
b = a.view(dtype=int)
b[0] = 10
assert a[0] == b[0]
```

Reply:
definitely needs to be no-copy; i'll add that to the docstring

Reply:
there is a test for the no-copy bit